packages/zapp/console/src/components/Launch/LaunchForm/UnionInput.tsx

```diff
@@ -0,0 +1,156 @@
+import { Typography } from '@material-ui/core';
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import * as React from 'react';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import { InputProps, InputType, InputTypeDefinition, UnionValue, InputValue } from './types';
+import { formatType } from './utils';
+import { getComponentForInput } from './LaunchFormInputs';
+import { getHelperForInput } from './inputHelpers/getHelperForInput';
+import { SearchableSelector, SearchableSelectorOption } from './SearchableSelector';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  inlineTitle: {
+    display: 'flex',
+    gap: theme.spacing(1),
+    alignItems: 'center',
+    paddingBottom: theme.spacing(3),
+  },
+}));
+
+const generateInputTypeToValueMap = (
+  listOfSubTypes: InputTypeDefinition[] | undefined,
+  initialInputValue: UnionValue | undefined,
+  initialType: InputTypeDefinition,
+): Record<InputType, InputValue> | {} => {
+  if (!listOfSubTypes?.length) {
+    return {};
+  }
+
+  return listOfSubTypes.reduce(function (map, subType) {
+    if (initialInputValue && subType === initialType) {
+      map[subType.type] = initialInputValue;
+    } else {
+      map[subType.type] = { value: '', typeDefinition: subType };
+    }
+    return map;
+  }, {});
+};
+
+const generateSearchableSelectorOption = (
+  inputTypeDefinition: InputTypeDefinition,
+): SearchableSelectorOption<InputType> => {
+  return {
+    id: inputTypeDefinition.type,
+    data: inputTypeDefinition.type,
+    name: formatType(inputTypeDefinition),
+  } as SearchableSelectorOption<InputType>;
+};
+
+const generateListOfSearchableSelectorOptions = (
+  listOfInputTypeDefinition: InputTypeDefinition[],
+): SearchableSelectorOption<InputType>[] => {
+  return listOfInputTypeDefinition.map((inputTypeDefinition) =>
+    generateSearchableSelectorOption(inputTypeDefinition),
+  );
+};
+
+export const UnionInput = (props: InputProps) => {
+  const { initialValue, required, label, onChange, typeDefinition, error, description } = props;
+
+  const classes = useStyles();
+
+  const listOfSubTypes = typeDefinition?.listOfSubTypes;
+
+  if (!listOfSubTypes) {
+    return <></>;
+  }
+
+  const inputTypeToInputTypeDefinition = listOfSubTypes.reduce(
+    (previous, current) => ({ ...previous, [current.type]: current }),
+    {},
+  );
+
+  const initialInputValue =
+    initialValue &&
+    (getHelperForInput(typeDefinition.type).fromLiteral(
+      initialValue,
+      typeDefinition,
+    ) as UnionValue);
+
+  const initialInputTypeDefinition = initialInputValue?.typeDefinition ?? listOfSubTypes[0];
```

Comment:
can we at this point guarantee that listOfSubTypes has at least one value (and not an empty array [])?

Reply:
yes checked the value and the length in the beginning 
https://github.com/flyteorg/flyteconsole/blob/866e71bac816085683611c907aa452035d3b3ee6/packages/zapp/console/src/components/Launch/LaunchForm/UnionInput.tsx#L66